lib/ansible/cli/pull.py

```diff
@@ -255,6 +256,17 @@ def run(self):
             display.display("Repository has not changed, quitting.")
             return 0
 
+        # Install Galaxy Requirements
+        requirements = self.select_requirements(context.CLIARGS['dest'])
+        if requirements is not None:
+            cmd = '%s/ansible-galaxy install --force --force-with-deps --role-file %s' % (bin_path, requirements)
```

Comment:
Using `--force` and `--force-with-deps` seems like it should be optional. Not exposing the options to the user makes this limited. Why not create a setup playbook and pass both playbooks `ansible-pull setup.yml playbook.yml ...`? Then you could still run one command but tailor the install process. For example, someone may just want to upgrade collections. Or only install a role isn't already available. Or specify a different install destination.

Reply:
Yes, I agree. I was a bit hasty. I've put them behind the `--force` and `--galaxy-force-with-deps` flags.

As for the reason I decided to do this instead of having a setup.yml file is that I use the `select_playbook` functionality lower in the file. This autoselects the playbook.yml based on hostname. If I pass a setup.yml, it doesn't autoselect the hostname.yml anymore.

Reply:
I've tried that option. I ended up with a setup playbook that looks something like this:
```yml
- hosts: localhost
  become: yes
  tasks:
    - name: Run Ansible Galaxy
      community.general.ansible_galaxy_install:
        requirements_file: roles/requirements.yml
        type: both
        force: yes
```

The anoying this is (and my reason for making this PR) that this requires you to run `ansible-galaxy collection install community.general` first. Whereas the method in this PR means you just have to run ansible-pull. No setup required.

Another alternative could obviously be to run the command module as such:
```yml
- hosts: localhost
  become: yes
  tasks:
    - name: Run Ansible Galaxy
      ansible.builtin-command:
        cmd: /usr/bin/ansible-galaxy --force-with-deps --role-file roles/requirements.yml
```

But that just seems like a bad practice, seeing how there is a module available.
I'd love to hear your thoughts on this.

For reference, see this page: https://docs.ansible.com/ansible/latest/collections/community/general/ansible_galaxy_install_module.html#ansible-collections-community-general-ansible-galaxy-install-module

Reply:
> I've tried that option.

It doesn't exist. I was suggesting adding a new option to allow that to be done with a single `ansible-pull` command.
In the playbook, I meant using `ansible-galaxy` directly as you did in the alternative approach. I wasn't aware of the module.

Keeping with the convention already used by awx/controller seems like a better idea though. It would be good if collections worked the same way. For what it's worth, it also looks like awx does not use `--force` or `--force-with-deps`, and there is no toggle.